### DISCUSSION TITLE
Using spritecode in string

### DISCUSSION THREAD

**Discussion by Batcastle** [Monika-After-Story/MonikaModDev#8315] [QA - code and content creation (technical)] [answered]
I am trying to write a submod for MAS and have run into a snag.

I have JSON file, with formatting similar to this:
```json
{
    "M": {
                 "element1":[["spritecode", "dialog"],
                             ["spritecode", "dialog"]]
          }
}
```
After reading it in and parsing it down based on a user's information, I end up with this in a var:
```json
[
    ["spritecode", "dialog"],
    ["spritecode", "dialog"]
]
```
The idea WAS that I could do something like the following (assuming the above data is in a var named `var`) and have easily variable dialog based on the individual user:
```python
for each in var:
    m each[0] "[each[1]]"
```
Unfortunately, that doesn't work cause apparently Ren'Py doesn't have for loops (????). I was able to rewrite to this fairly easily:
```ren'py
$ each = 0
while each < len(var):
    m var[each][0] "[var[each][1]]"
    $ each = each + 1

```
At this point, I realized that spritecodes are not strings, they are something else. So if I try to pass `var[each][0]` to `m`, it will crash more than likely.

So, comes my question: is there a function in the Python environment I can use to set Monika's expression using a spritecode stored in a string? What about setting dialog? If that's not possible in Python, what about in native Ren'Py?

Thanks ahead of time!

**Comment by Booplicate** [comment, marked as answer]
Yep, renpy doesn't have `for` loop in its syntax. You can use Python blocks for them, like:
```renpy
label test:
    python:
        for i in range(9)
            print(str(i))
    return
```

Spritecodes are strings, but renpain doesn't support dynamic spritecodes. You'll have to use Python equivalents
```renpy
label test:
    python:
        renpy.show("spritecode")
        renpy.say(m, "dlg")
    return
```
Which may be slower, by the way.
Alternatively
```renpy
label test:
    show expression spritecode
    m "[dlg]"
    return
```
Which should be ok, but less clear to read.

**Reply by Batcastle** [reply]
While `renpy.say()` works like a charm, `renpy.show()` isn't doing anything. Monika's expression doesn't change. Is there something more than just the spritecode I need to pass? According the [the Ren'Py documentation i found](https://www.renpy.org/doc/html/displaying_images.html) I shouldn't call `renpy.hide()` before hand, but do I need to ignore that?

**Reply by Booplicate** [reply]
You don't need `renpy.hide` for this, renpy will replace the image since they have the same tag. Which I assume you forgot to add. For the function you have to use full image names, `"monika 1eua"` as an example.

**Reply by Batcastle** [reply]
Ohhhh yeah I was missing the "monika" part. That got it working. Thank you so much!
